functions/src/cron-jobs/slack.ts

```diff
@@ -0,0 +1,58 @@
+import * as functions from 'firebase-functions';
+import { SLACK_SECRET_KEYS } from '../constants/secretKeys';
+import { FORBIDDEN_MEMBERS, MESSAGE_SHORT, MESSAGE_FULL, TIMEZONE } from '../constants/slack';
+import { getSlackClient } from '../services/slack';
+import { getSecrets } from '../utils/getSecrets';
+
+const getSlackUsersIds = async () => {
+  const { SLACK_TOKEN } = getSecrets();
+  const slack = getSlackClient(SLACK_TOKEN);
+  const usersList = await slack.users.list();
+  if (!usersList.members) return [];
+
+  const filteredList = usersList.members.filter((member) => {
+    return (
+      !member.is_bot &&
+      !member.is_restricted &&
+      !member.is_ultra_restricted &&
+      !member.is_invited_user &&
+      !member.deleted &&
+      !FORBIDDEN_MEMBERS.includes(member.id || '')
+    );
+  });
+
+  const usersIdsList = filteredList.map((user) => user.id);
+
+  return usersIdsList as string[];
+};
+
+const sendMessageToUsers = async () => {
+  const { SLACK_TOKEN } = getSecrets();
+  const slack = getSlackClient(SLACK_TOKEN);
+  const userIds = await getSlackUsersIds();
```

Comment:
Maybe I'm wrong but I think this function also should have its own `try/catch`

Reply:
I think this combination will go well 
```js
if (usersList.error) console.error('Failed to fetch slack users data');
if (!usersList.members) return [];
```
WDYT?

Reply:
yup, should be good enogh